httpserver/swarm.cpp

```diff
@@ -134,36 +134,46 @@ void Swarm::set_swarm_id(swarm_id_t sid) {
     cur_swarm_id_ = sid;
 }
 
-static std::unordered_map<std::string, sn_record_t>
-get_snode_map_from_swarms(const all_swarms_t& swarms) {
+static auto get_snode_map_from_swarms(const all_swarms_t& swarms) {
 
-    std::unordered_map<std::string, sn_record_t> snode_map;
+    std::unordered_map<legacy_pubkey, sn_record_t> snode_map;
     for (const auto& swarm : swarms) {
         for (const auto& snode : swarm.snodes) {
-            snode_map.insert({snode.sn_address(), snode});
+            snode_map.emplace(snode.pubkey_legacy, snode);
         }
     }
     return snode_map;
 }
 
+template <typename T>
+bool update_if_changed(T& val, const T& new_val, const std::common_type_t<T>& ignore_val) {
+    if (new_val != ignore_val && new_val != val) {
+        val = new_val;
+        return true;
+    }
+    return false;
+}
+
 auto apply_ips(const all_swarms_t& swarms_to_keep,
                const all_swarms_t& other_swarms) -> all_swarms_t {
 
     all_swarms_t result_swarms = swarms_to_keep;
     const auto other_snode_map = get_snode_map_from_swarms(other_swarms);
 
     int updates_count = 0;
-    for (auto& swarm : result_swarms) {
-        for (auto& snode : swarm.snodes) {
+    for (auto& [swarm_id, snodes] : result_swarms) {
+        for (auto& snode : snodes) {
             const auto other_snode_it =
-                other_snode_map.find(snode.sn_address());
+                other_snode_map.find(snode.pubkey_legacy);
             if (other_snode_it != other_snode_map.end()) {
-                const auto& new_ip = other_snode_it->second.ip();
-                // Keep swarms_to_keep but don't overwrite with default IPs
-                if (new_ip != "0.0.0.0" && snode.ip() != new_ip) {
-                    snode.set_ip(new_ip);
+                auto& sn = other_snode_it->second;
+                // Keep swarms_to_keep but don't overwrite with default IPs/ports
+                bool updated = false;
+                updated += update_if_changed(snode.ip, sn.ip, "0.0.0.0");
```

Comment:
i know that addition on `bool` is valid it just feels kind of icky

Reply:
Changed to:
```
if (update_if_changed(...)) updated = true;
if (update_if_changed(...)) updated = true;
```
etc.